libcst/metadata/scope_provider.py

```diff
@@ -616,7 +616,10 @@ def _gen_dotted_names(
             # this is not an import
             return
         name_values = iter(_gen_dotted_names(value))
-        (next_name, next_node) = next(name_values)
+        next_pair = next(name_values, None)
+        if next_pair is None:
+            return
```

Comment:
Can we just add `try ... except StopIteration`? That can be more efficient.

Reply:
Hmm, sure. Why is it more efficient?

Reply:
That way we saved one extra `next` call.

Reply:
Then I don't understand your suggestion :)
Apart from the way it's currently written, I can imagine this:
```
try:
    (next_name, next_node) = next(name_values)
    yield (f"...", ...)
    ...
except StopIteration:
    return
```

which is the same amount of `next` calls